src/coreclr/src/jit/hwintrinsicxarch.cpp

```diff
@@ -1501,6 +1501,38 @@ GenTree* Compiler::impBMI1OrBMI2Intrinsic(NamedIntrinsic        intrinsic,
 
             if (sig->numArgs == 3)
             {
+                // If op3 (low part of result) is an address of a local variable of the correct type:
+                //   MultiplyNoFlags(*, *, &someLocal);
+                // Then it is transformed to use a dedicated local for that whose value is immediately copied to the
+                // original op3 local after MultiplyNoFlags (but before producing the main result):
+                //   MultiplyNoFlags(*, *, &op3TempLocal), MOV(someLocal, op3TempLocal);
+                //
+                // After this the original local isn't marked as address-exposed and lowering will see a GT_LCL_VAR
+                // node for the op3TempLocal immediately after MULX (as GT_COMMA is transformed right before lowering)
+                // which enables it to remove the temp local entirely and replace its use with a special value node.
+                if (opts.OptimizationEnabled() && op3->OperGet() == GT_ADDR &&
+                    op3->gtGetOp1()->OperGet() == GT_LCL_VAR && op3->gtGetOp1()->TypeGet() == callType)
+                {
+                    // Get a single-use temp to store the 2nd value which we'll later remove in lowering.
+                    GenTreeLclVar* op3Local = op3->gtGetOp1()->AsLclVar();
+                    const unsigned orgNum   = op3Local->GetLclNum();
+                    const unsigned tmpNum   = lvaGrabTemp(true DEBUGARG("MULX low part"));
+                    lvaTable[tmpNum].lvType = callType;
+                    op3Local->SetLclNum(tmpNum);
+
+                    GenTree* mulx = gtNewScalarHWIntrinsicNode(callType, op1, op2, op3, intrinsic);
+
+                    // Move the 2nd value to the original local var and stitch it together. This transformation produces
+                    // correct results even if lowering fails to optimize it away.
+                    GenTree* mov = gtNewAssignNode(gtNewLclvNode(orgNum, callType), gtNewLclvNode(tmpNum, callType));
+
+                    GenTree* comma = gtNewOperNode(GT_COMMA, callType, mov, mulx);
+                    comma->gtFlags |= GTF_REVERSE_OPS;
```

Comment:
My understanding is that `mulx` must be the last op for `COMMA` because it will be the value produced from it. But they must be executed in the reverse order.